src/algebra/big_operators/fin.lean

```diff
@@ -161,6 +161,22 @@ end partial_prod
 
 end fin
 
+namespace vector
+
+lemma sum_ite_eq_nth_eq_count [decidable_eq α] [add_comm_monoid_with_one β] {n : ℕ}
+  (a : α) (v : vector α n) : ∑ i, ite (a = v.nth i) (1 : β) 0 = ↑(v.to_list.count a) :=
+begin
+  induction n with n hn,
+  { simp [vector.eq_nil v] },
+  { obtain ⟨x, xs, hxs⟩ := v.exists_eq_cons,
+    suffices : ite (x = a) (1 : β) 0 + xs.to_list.count a =
+      ite (x = a) ((xs.to_list.count a) + 1) (xs.to_list.count a),
+    by simpa [hxs, hn xs, fin.sum_univ_succ, list.count_cons, @eq_comm _ a x] using this,
+    split_ifs; simp [add_comm (1 : β)] }
```

Comment:
I think you might find this easier to prove if you start with
```
induction v using vector.induction_on
```

Reply:
I just used the function directly, but yes this cleaned up the proof a good bit

Reply:
You're right, the spelling above doesn't work. With #15383, `induction v using vector.induction` will work.